### ISSUE TITLE
SSL Certs Reload with Refresh event not working in v3.6.0?

### ISSUE THREAD

**Issue by anuragdeshpande** [micronaut-projects/micronaut-core#7873]
Outside of official docs, I am unable to find any references online for this config. So responding to the merge where the feature was added. 

@graemerocher Thank you so much for adding this. It would really help us out in our application. I am on micronaut v3.6.0, gradle 7.3, java 11 project. However, I don't seem to be able to get this config to work. 

Intent: I get new .p12 files everytime the certs are renewed. Without restarting the service, I would like to refresh the keymanagement system in the micronaut server to use the new .p12 file (public key value changes from the previous one in use). 

I have tried several versions and even tried passing in the full file path. It still seems to be using the file that was included in the original build. I am attaching screenshots of my config, attempts and logs. Am I missing anything? Could you please help? 

Original SSL config in application
![image](https://user-images.githubusercontent.com/3684197/184462424-5e19179b-1c6b-4c40-91c0-e4fea1ed04de.png)

Logging on refresh event
![image](https://user-images.githubusercontent.com/3684197/184462212-6b0c9cde-695b-4973-a081-df485a1930ed.png)

Trying full ssl.* reload
![image](https://user-images.githubusercontent.com/3684197/184462504-273aa722-a04b-4055-a90e-2252e12d4c82.png)

Confirming refresh event in logs
![image](https://user-images.githubusercontent.com/3684197/184462093-154d7567-c97a-439d-a1f8-db81d893c703.png)

Application returning old value in property as response. Also tried new requests in a new browser window, different browser as well
![image](https://user-images.githubusercontent.com/3684197/184461914-4dea3d85-f5d6-4aa0-abfe-30385ab1d9cf.png)

Trying just path refresh for SSL. Event was successfully recorded in logs, but still not using new p12 file.
![image](https://user-images.githubusercontent.com/3684197/184463061-5afaa4c7-47d4-4d08-8d97-cd86c209f571.png)

_Originally posted by @anuragdeshpande in https://github.com/micronaut-projects/micronaut-core/issues/6112#issuecomment-1213625246_

**Comment by anuragdeshpande** [comment]
First I assumed, a new file cannot be injected. Instead only the contents of the existing file to add the new cert. However, as expected, OS has a lock on the file so I cannot edit, update or delete without bringing the service down.

I have also tried mapping the cert files in docker containers to check if it was a windows/ file system thing. I am seeing same behavior in linux containers as well. 

My hope is that we can map a new p12 file and trigger refresh event to point at application the new file and reload cert without needing a restart.

PS: As indicated in the documentation, the intent is to write a scheduled method. I have used a controller just for this post so as to get more control on simulation and take screenshots.

**Comment by anuragdeshpande** [comment]
@graemerocher I noticed that values are dropped in refresh event and only the keySet is used to refresh the application. If we track the values as well and use them to inject the updated bean instead, we can reload all config with dynamic values at runtime. Would this be a security implication if implemented at core framework level?

![image](https://user-images.githubusercontent.com/3684197/184508653-ed919e9e-4fd7-4482-8d9b-e2250cc3ba4c.png)

I was able to hot reload a file using refresh event by injecting the SSL Configuration bean. After a bunch of experiments this is what worked for me. 

![image](https://user-images.githubusercontent.com/3684197/184508553-6f8ed44d-0796-4b82-b43a-4ed055423860.png)
